spirv_msl.cpp

```diff
@@ -1253,7 +1253,16 @@ string CompilerMSL::entry_point_args(bool append_comma)
 				case SPIRType::Struct:
 					if (!ep_args.empty())
 						ep_args += ", ";
-					ep_args += "constant " + type_to_glsl(type) + "& " + to_name(var.self);
+					if ((meta[type.self].decoration.decoration_flags & (1ull << DecorationBufferBlock)) != 0 ||
+						(meta[type.self].decoration.decoration_flags & (1ull << DecorationNonWritable)) == 0)
```

Comment:
Just to be sure, you are suggesting changing the || in a && because only BufferBlock can be writable?

From the MSL spec:

MSL 1.2 spec, section 4.2.3
> The constant address space name refers to buffer memory objects allocated from the device memory
pool but are read-only. 

MSL 1.2 spec, section 4.2.1
> The device address space name refers to buffer memory objects allocated from the device memory
pool that are both readable and writeable. 

MSL doesn't seem to make (or expose) a distinction between different memory types, constant is only used to tag something as readonly.